src/soundbase.cpp

```diff
@@ -287,15 +339,35 @@ printf ( "\n" );
                     // make sure packet is long enough
                     if ( vMIDIPaketBytes.Size() > 2 )
                     {
-                        // we are assuming that the controller number is the same
-                        // as the audio fader index and the range is 0-127
-                        const int iFaderLevel = static_cast<int> ( static_cast<double> (
-                            qMin ( vMIDIPaketBytes[2], uint8_t ( 127 ) ) ) / 127 * AUD_MIX_FADER_MAX );
-
-                        // consider offset for the faders
-                        const int iChID = vMIDIPaketBytes[1] - iMIDIOffsetFader;
-
-                        emit ControllerInFaderLevel ( iChID, iFaderLevel );
+                        const auto &cCtrl = aMidiCtls[qMin ( vMIDIPaketBytes[1], uint8_t ( 127 ) )];
```

Comment:
Irritating style issue... I _think_ (not sure) it should be `const auto& ....` - the `&` being the type not the name.

I'm also unsure on the use of `auto` generally.

Finally, I'm a bit dubious about using the qMin() to determine the index into the array.  I'd have expected an out-of-bounds value to be rejected.  (No need for any message - but it's an invalid request so shouldn't go into the switch.)

Reply:
`auto` of course is standard C++11.  If you don't want to see it used, it would make sense putting it in the list of style mandates for working on Jamulus.